modules/kafka/kafka_helpers_test.go

```diff
@@ -9,101 +9,171 @@ import (
-				ContainerRequest: testcontainers.ContainerRequest{
-					NetworkAliases: map[string][]string{
-						"foo":  {"foo0", "foo1", "foo2", "foo3"},
-						"bar":  {"bar0", "bar1", "bar2", "bar3"},
-						"baaz": {"baaz0", "baaz1", "baaz2", "baaz3"},
-					},
+		}, "1@foo0:9094")
+	})
+
+	t.Run("voters/localhost/alias-no-networks", func(t *testing.T) {
+		testConfigureControllerQuorumVotersFn(t, &testcontainers.GenericContainerRequest{
+			ContainerRequest: testcontainers.ContainerRequest{
+				NetworkAliases: map[string][]string{
+					"foo":  {"foo0", "foo1", "foo2", "foo3"},
+					"bar":  {"bar0", "bar1", "bar2", "bar3"},
+					"baaz": {"baaz0", "baaz1", "baaz2", "baaz3"},
 				},
 			},
-			expectedVoters: "1@localhost:9094",
-		},
-	}
+		}, "1@localhost:9094")
+	})
+}
 
-	for _, test := range tests {
-		t.Run(test.name, func(t *testing.T) {
-			configureControllerQuorumVoters(test.req)
+func TestValidateKRaftVersion(t *testing.T) {
+	validateKRaftVersionFn := func(t *testing.T, image string, wantErr bool) {
+		t.Helper()
+
+		err := validateKRaftVersion(image)
 
-			require.Equalf(t, test.expectedVoters, test.req.Env["KAFKA_CONTROLLER_QUORUM_VOTERS"], "expected KAFKA_CONTROLLER_QUORUM_VOTERS to be %s, got %s", test.expectedVoters, test.req.Env["KAFKA_CONTROLLER_QUORUM_VOTERS"])
-		})
+		if wantErr {
+			require.Error(t, err)
+		} else {
+			require.NoError(t, err)
+		}
 	}
+
+	t.Run("official/valid-version", func(t *testing.T) {
+		validateKRaftVersionFn(t, "confluentinc/confluent-local:7.5.0", false)
+	})
+
+	t.Run("official/valid-limit-version", func(t *testing.T) {
+		validateKRaftVersionFn(t, "confluentinc/confluent-local:7.4.0", false)
+	})
+
+	t.Run("official/invalid-low-version", func(t *testing.T) {
+		validateKRaftVersionFn(t, "confluentinc/confluent-local:7.3.99", true)
+	})
+
+	t.Run("official/invalid-too-low-version", func(t *testing.T) {
+		validateKRaftVersionFn(t, "confluentinc/confluent-local:5.0.0", true)
+	})
+
+	t.Run("unofficial/does-not-validate-KRaft-version", func(t *testing.T) {
+		validateKRaftVersionFn(t, "my-kafka:1.0.0", false)
+	})
 }
 
-func TestValidateKRaftVersion(t *testing.T) {
-	tests := []struct {
-		name    string
-		image   string
-		wantErr bool
-	}{
-		{
-			name:    "Official: valid version",
-			image:   "confluentinc/confluent-local:7.5.0",
-			wantErr: false,
-		},
-		{
-			name:    "Official: valid, limit version",
-			image:   "confluentinc/confluent-local:7.4.0",
-			wantErr: false,
-		},
-		{
-			name:    "Official: invalid, low version",
-			image:   "confluentinc/confluent-local:7.3.99",
-			wantErr: true,
-		},
-		{
-			name:    "Official: invalid, too low version",
-			image:   "confluentinc/confluent-local:5.0.0",
-			wantErr: true,
-		},
-		{
-			name:    "Unofficial does not validate KRaft version",
-			image:   "my-kafka:1.0.0",
-			wantErr: false,
-		},
+func TestTrimValidateListeners(t *testing.T) {
+	validateFn := func(t *testing.T, listeners []KafkaListener, wantErr bool) {
```

Comment:
suggestion: as above, eliminate the func and embed directly to make it clearer as true / false are is ambiguous until you look at the function.